sdk/tests/conformance2/transform_feedback/transform_feedback_same_buffers_as_input_and_output.html

```diff
@@ -0,0 +1,153 @@
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL Transform Feedback Conformance Test - Same Buffers as Input and Output</title>
+<link rel="stylesheet" href="../../resources/js-test-style.css"/>
+<script src="../../js/js-test-pre.js"></script>
+<script src="../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<canvas id="canvas" style="width: 50px; height: 50px;"> </canvas>
+<div id="console"></div>
+<script id="vshader" type="x-shader/x-vertex">#version 300 es
+in float in_value;
+out float out_value;
+
+void main() {
+   out_value = in_value * 2.;
+}
+</script>
+<script id="fshader" type="x-shader/x-fragment">#version 300 es
+precision mediump float;
+out vec4 dummy;
+void main() {
+  dummy = vec4(1);
+}
+</script>
+<script>
+"use strict";
+description("This test verifies trying to read and write from the same buffers fails");
+
+debug("");
+
+var wtu = WebGLTestUtils;
+var canvas = document.getElementById("canvas");
+var gl = wtu.create3DContext(canvas, null, 2);
+
+if (!gl) {
+    testFailed("WebGL context does not exist");
+} else {
+    testPassed("WebGL context exists");
+
+    runSameInputOutputTest();
+    finishTest();
+}
+
+function runSameInputOutputTest() {
+  const prog = wtu.setupTransformFeedbackProgram(gl, ["vshader", "fshader"],
+      ["out_value"], gl.SEPARATE_ATTRIBS,
+      ["in_value"]);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "linking transform feedback shader should not set an error");
+
+  const inLoc = 0;
+  const outLoc = 0;
+
+  const srcBuffer1 = createBuffer(gl, new Float32Array([1, 2, 3]));
+  const srcVAO1 = createVAO(gl, srcBuffer1, inLoc);
+
+  const dstBuffer = createBuffer(gl, Float32Array.BYTES_PER_ELEMENT * 3);
+  const srcVAO2 = createVAO(gl, dstBuffer, inLoc);
+
+  const tf = gl.createTransformFeedback();
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+  gl.useProgram(prog);
+  gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, dstBuffer);
+  // this binds the default (id = 0) TRANSFORM_FEEBACK buffer
+  gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null);
+
+  runFeedback(gl, prog, srcVAO1, tf);
+  wtu.glErrorShouldBe(gl, gl.NO_ERROR, "there should be no errors");
+
+  const expected = [2, 4, 6];
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected);
+
+  // this should fail because the transform feedback's buffer #0 and the
+  // srcVAO2's buffer #0 are the same buffer
+  runFeedback(gl, prog, srcVAO2, tf);
+  wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "reading and writing to same buffer");
+
+  gl.bindBuffer(gl.ARRAY_BUFFER, dstBuffer);
+  wtu.checkFloatBuffer(gl, gl.ARRAY_BUFFER, expected, "should be the same as before as nothing has executed");
```

Comment:
If incorrectly executed (as it currently is in chrome) the result is 4, 8, 12 because it's manipulating the result from the first `* 2.0` operation.

That result is probably GPU dependent since the whole problem is results of having the same input and output buffer is undefined. In any case it shouldn't execute and the buffer should still have 2, 4, 6 in it. It doesn't in Chrome currently on my MBP. Seems like the test is valid